src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -168,5 +167,18 @@ private bool CloseHandle(bool abortive, bool canceledOperations)
             }
 #endif
         }
+
+        private new void SetHandle(IntPtr handle)
```

Comment:
I've refactored the Windows/Unix/ctor to all use this common function for checking `IsInvalid`.